app/src/lib/components/Carousel/Carousel.tsx

```diff
@@ -1,60 +1,98 @@
-  slideToShow = 1,
-  isArrowShow = true,
-  isAutoplay = false,
-}: Props) => {
-  const [showIndex, setShowIndex] = useState<number>(0);
-
-  const childrenLen = useMemo(() => React.Children.toArray(children).length, [children]);
-
-  const showPrev = () => {
-    if (showIndex === 0) return setShowIndex(() => Math.floor((childrenLen - 1) / slideToShow));
-    setShowIndex((prev) => prev - 1);
-  };
-
-  const showNext = () => {
-    if (showIndex === Math.floor((childrenLen - 1) / slideToShow)) return setShowIndex(() => 0);
-    setShowIndex((prev) => prev + 1);
-  };
-
-  isAutoplay && useInterval(showNext, autoplaySpeed, [showIndex]);
-
-  return (
-    <Wrapper len={childrenLen} transition={transition} showIndex={showIndex}>
-      {isArrowShow && <FaArrowCircleLeft id="prev-button" onClick={showPrev} />}
-      <div className="carousel-wrapper">
-        <div className="carousel-container">
-          {React.Children.map(children, (child) => {
-            return (
-              <Container len={childrenLen} slideToShow={slideToShow} key={child?.toString()}>
-                {child}
-              </Container>
-            );
-          })}
-        </div>
-      </div>
-      {isArrowShow && <FaArrowCircleRight id="next-button" onClick={showNext} />}
-    </Wrapper>
-  );
+type ArrowLocationType = {
+  top?: string;
+  bottom?: string;
+  side?: string;
+  translateY?: string;
 };
 
+const Carousel = React.forwardRef(
+  (
+    {
+      children,
+      width,
+      padding,
+      transition = 1000,
+      autoplaySpeed = 3000,
+      slideToShow = 1,
+      isArrowShow = true,
+      isAutoplay = false,
+      arrowLocation = 'mid-side',
+      prevArrowIcon = <FaArrowCircleLeft />,
+      nextArrowIcon = <FaArrowCircleRight />,
+    }: Props,
+    ref: React.ForwardedRef<HTMLDivElement>
+  ) => {
+    const [showIndex, setShowIndex] = useState<number>(0);
+
+    const childrenLen = useMemo(() => React.Children.toArray(children).length, [children]);
+    const lastChildIndex = useMemo(() => Math.floor((childrenLen - 1) / slideToShow), [childrenLen, slideToShow]);
+
+    const showPrev = () => {
+      if (showIndex === 0) return setShowIndex(() => lastChildIndex);
+      setShowIndex((prev) => prev - 1);
+    };
+
+    const showNext = () => {
+      if (showIndex === lastChildIndex) return setShowIndex(() => 0);
+      setShowIndex((prev) => prev + 1);
+    };
+
+    /* These const variables are ArrowIcons received to props */
+    const sizedPrevArrowIcon = useMemo(() => cloneElement(prevArrowIcon), [prevArrowIcon]);
+    const sizedNextArrowIcon = useMemo(() => cloneElement(nextArrowIcon), [nextArrowIcon]);
+
+    /* useInterval is setTimeout custom hook */
+    isAutoplay && useInterval(showNext, autoplaySpeed, [showIndex]);
+
+    return (
+      <Wrapper arrowLocation={arrowLocation} width={width} padding={padding}>
+        {isArrowShow && (
+          <div className="icon-wrapper" id="prev-button" onClick={showPrev}>
```

Comment:
className 사용하지 않으면 지웁시당
```suggestion
          <div id="prev-button" onClick={showPrev}>
```
```suggestion
          <div className="icon-wrapper" id="prev-button" onClick={showPrev}>
```

Reply:
저기서 className을 넣은건 해당 div의 역할을 명시해주려고 넣은거에요 그래도 필요 없어보이면 뺄게요